src/goto-programs/remove_function_pointers.h

```diff
@@ -46,4 +47,32 @@ bool remove_function_pointers(
   bool add_safety_assertion,
   bool only_remove_const_fps = false);
 
+/// Replace a call to a dynamic function at location
+/// target in the given goto-program by a case-split
+/// over a given set of functions
+/// \param message_handler: Message handler
+/// \param symbol_table: Symbol table
+/// \param goto_program: The goto program that contains target
+/// \param function_id: Name of function containing the target
+/// \param target: location with function call with function pointer
+/// \param functions: The set of functions to consider
+/// \param add_safety_assertion: Iff true, include an assertion that the
+//         pointer matches one of the candidate functions
+void remove_function_pointer(
+  message_handlert &message_handler,
+  symbol_tablet &symbol_table,
+  goto_programt &goto_program,
+  const irep_idt &function_id,
+  goto_programt::targett target,
+  const std::unordered_set<symbol_exprt, irep_hash> &functions,
```

Comment:
This is the key line right?  It gives a way of passing a map to function pointer removal and this splitting the process into "generate function maps" and "convert code using the maps"?